classes/class.ilOpencastEventConfigGUI.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * ilOpencastEventConfigGUI
+ *
+ * @author Farbod Zamani Boroujeni <zamani@elan-ev.de>
+ * @extends ilPluginConfigGUI
+ */
+class ilOpencastEventConfigGUI extends ilPluginConfigGUI
+{
+    /**
+     * @var elanev\OpencastEvent\Config\PluginConfig
+     */
+    protected $config_object;
+    /**
+     * @var \ilCtrlInterface
+     */
+    private $ctrl;
+    /**
+     * @var \ilGlobalTemplateInterface
+     */
+    private $main_tpl;
+    /**
+     * @var \ilLanguage
+     */
+    private $language;
+    /**
+     * @var \ilTabsGUI
+     */
+    private $tabs;
+
+    public function __construct()
+    {
+        global $DIC;
+        $this->ctrl = $DIC->ctrl();
+        $this->main_tpl = $DIC->ui()->mainTemplate();
+        $this->language = $DIC->language();
+        $this->tabs = $DIC->tabs();
+        $this->config_object = new \elanev\OpencastEvent\Config\PluginConfig();
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getFields()
+    {
+        return [
+            $this->config_object::F_THUMBNAIL_LINK => [
+                'type' => 'ilCheckboxInputGUI',
+                'hasInfo' => true,
+                'subItems' => [],
+                'value' => (bool) $this->config_object::getConfig($this->config_object::F_THUMBNAIL_LINK) ?? false
+            ]
+        ];
+    }
+
+    public function executeCommand(): void
+    {
+        parent::executeCommand();
+    }
+
+    public function performCommand($cmd): void
+    {
+        switch ($cmd) {
+            case "configure":
+            case "save":
+                $this->$cmd();
+                break;
+        }
+    }
+
+    /**
+     * Configure screen
+     */
+    public function configure(): void
+    {
+        $this->initConfigurationForm();
+        $this->fillForm();
+        $this->main_tpl->setContent($this->form->getHTML());
+    }
+
+    /**
+     * Set form item values
+     */
+    private function fillForm(): void
+    {
+        $values = [];
+        foreach ($this->getFields() as $key => $item) {
+            if (isset($item['value'])) {
+                $values[$key] = $item['value'];
+            }
+        }
+        $this->form->setValuesByArray($values);
+    }
+
+
+    /**
+     * @return ilPropertyFormGUI
+     */
+    public function initConfigurationForm(): \ilPropertyFormGUI
+    {
+        $this->form = new ilPropertyFormGUI();
```

Comment:
I can see that these are all quite simple inputs (checkbox, for example). I would therefore recommend implementing them directly with the new forms from the UI service. The previous forms (ilPropertyFormGUI) were abolished at some point with ILIAS 10/11/..., so it makes sense to write new forms with the new service now. I can show you how this works if you still want to do this, but of course it can also be done later.

Reply:
Great, of course, we will make this change, and I will ask for your help later on. Thanks